### ISSUE TITLE
[Question] Monitor overclocking availability

### ISSUE THREAD

**Issue by hfc2x** [pop-os/cosmic-comp#743]
I'm not sure if this is the right place to ask this, but I just want to know if cosmic-comp will support monitor overclocking?

I ask because it's something that some monitors support officially (for example certain ASUS monitors), but it's something the compositor has to also support.

Back in the X11 days, this could be done with the use of `xrandr` to create a custom configuration, but Wayland obviously doesn't support it.

Certain monitors don't support overclocking "officially", but `xrandr` still allows this to be done if the monitor's hardware is compatible with the selected refresh rate.

The only mention of this being a planned feature in a Wayland compositor is [this Reddit post](https://www.reddit.com/r/kde/comments/1cilvoi/comment/l2ks07i/), that mentions Kwin is supposedly getting support for this in the future. Unfortunately, it doesn't cite any sources, and I haven't been able to find anything official from the Kwin devs stating any of this.

I bring this up because in my main desktop, I run Fedora 40 KDE, with an ASUS VG259Q monitor, which supports 240Hz by default, and has an official 280Hz overclocking mode when using DisplayPort. I tried most of the solutions I could find online to get this to work, including setting specific kernel parameters at boot. Trying to get something to work led me to the Reddit post above.

So, I just want to know if cosmic-comp will allow monitor overclocking at some point? I'm not trying to make it sound like I want it to be available ASAP or anything like that. All I want is confirmation that there is at least some consideration for this in the future.

**Comment by Quackdoc** [comment]
wlr-randr supports setting custom resolutions, maybe that works? I don't have a monitor I can overclock but wlr-randr can be invoke like so `wlr-randr --output DP-1 --custom-mode 1920x1080@280`, not sure if this would work at all, but maybe it's worth a shot?

**Comment by hfc2x** [comment]
Yes, wlr-randr could theoretically be used for this, but it doesn't work with Kwin. This is because Kwin doesn't support `wlr-output-management-unstable-v1`, which I suppose the Reddit post is referring to.

So I suppose the question here would be if cosmic-comp will support this instead. Will update the title just to make sure.

**Comment by Quackdoc** [comment]
cosmic-comp supports wlr randr,  dunno about over clocking
